notebooks/simone_pipeline.md

```diff
@@ -0,0 +1,134 @@
+  2. If no, we will need some clean-up functions
+3. `max_projection` over z -> `starfish.image._stack` (currently present)
+4. Threshold finding. Is there an equivalent library function for the gradient calculation + trimming that this func does? -> `starfish.stats.calculate_peak_threshold_abs`. Replace `calculate` with your favorite word.
+5. `peak_local_max` -> `starfish.spots.peakfinder` (location from brian-long)
+
+## The pipeline:
+
+- First, `process_standalone_experiment.py` is called. 
+- This is primarily about munging files, and loads in `counting.filtering_and_counting` to do the work.
+
+### Arguments:
+
+1. path - path to experiment
+2. analysis name - prefix for experiment
+3. stringency (default 0) - ? TODO
+4. min-distance (default 5) - minimum distance between peaks
+5. min-plane (default None) - allows selection of a plane to start from (exclude out-of-focus)
+6. max-plane (default None) - allows selection of a plane to end at (exclude out-of-focus)
+
+- Next, `counting.filtering_and_counting` is called
+
+### Arguments:
+
+List Simone's, they're well documented. 
+
+```
+fpath_img_to_filter: str
+path to the file to process
+filtered_png_img_gene_dirs: list
+list of the paths of the directories where the filtered images as are
+saved as pngs.
+filtered_img_gene_dirs: list
+list of the paths of the directories where the filtered images are saved
+as .npy.
+counting_gene_dirs: list
+list of the paths of the directories where the countings of the filtered
+images are saved.
+illumination_correction: bool
+if True the illumination correction is run on the dataset.
+plane_keep: list
+start and end point of the z-planes to keep. Default None
+keep all the planes (ex. [2,-3]).
+min_distance: int
+minimum distance between dots. (peaks)
+stringency: int
+stringency use to select the threshold used for counting. (default 0)
+skip_genes_counting: list
+list of the genes to skip for counting count.
+skip_tags_counting: list
+list of the tags inside the genes/stainings name to avoid to count.
+```
+
+Procedure: 
+
+Note: Expects np.float64 images (how to address this? will everything work?)
+
+1. `filtering.nuclei_filtering`
+	1. called if this is in part of skip-genes, designed to skip nuclei, I think
+  2. 3d gaussian filter, sigma (2, 100, 100) (z, y, x) (could be z, x, y), but x <-> y for our purposes.
+  3. set negative values to zero
+  4. max project over z (np.amax(stack, axis=0)
+2. otherwise, call `filtering.smFISH_filtering`
+  1. `gaussian_filter`, sigma=(1, 8, 8)
+  2. `gaussian_laplace(stack, (0.2, 0.5, 0.5)`  # todo look up
+    3. this makes the peaks negative, so invert signal `img_stack = -img_stack`
+    4. ... and zero out the negative values again
```

Comment:
seems like a default parameter in the filters module, e.g., zero_negatives = False, could be flipped if the user wants to zero out negativ evalues